src/test/groovy/hudson/plugins/gradle/injection/BuildScanInjectionMavenIntegrationTest.groovy

```diff
@@ -503,6 +507,84 @@ node {
         getEnvVarFromNodeProperties(slave, MavenBuildScanInjection.JENKINSGRADLEPLUGIN_MAVEN_PLUGIN_CONFIG_EXT_CLASSPATH) == null
     }
 
+    @Unroll
+    @SuppressWarnings("GStringExpressionWithinString")
+    def 'vcs repository pattern injection for pipeline remote project - #pattern #mavenSetup'(String pattern, String mavenSetup) {
+        given:
+        withInjectionConfig {
+            injectionVcsRepositoryPatterns = pattern
+        }
+        createSlaveAndTurnOnInjection()
+        def mavenInstallationName = setupMavenInstallation()
+        def replacedMavenSetup = mavenSetup.replaceAll("mavenInstallationName", mavenInstallationName)
+
+        def pipelineJob = j.createProject(WorkflowJob)
+        pipelineJob.setDefinition(new CpsFlowDefinition("""
+   stage('Build') {
+        node('foo') {
+            $replacedMavenSetup
+                git branch: 'main', url: 'https://github.com/c00ler/simple-maven-project'
+                if (isUnix()) {
+                    sh "env"
+                    sh "mvn package -B"
+                } else {
+                    bat "set"
+                    bat "mvn package -B"
+                }
+            }
+        }
+   }
+""", false))
+
+        when:
+        // In this case, the job had to be executed twice for SCM to be populated (we do the same for Gradle, though
+        // for a different reason). Manual test didn't exhibit this problem, so most probably something caused by a test setup
+        j.buildAndAssertSuccess(pipelineJob)
+        def build = j.buildAndAssertSuccess(pipelineJob)
+
+        then:
+        if (pattern.contains("simple-")) {
+            j.assertLogContains("[INFO] The Gradle Terms of Service have not been agreed to.", build)
+        } else {
+            j.assertLogNotContains("[INFO] The Gradle Terms of Service have not been agreed to.", build)
+        }
+
+        where:
+        pattern                                     |  mavenSetup
+        "not-found-pattern, simple-"                | "withEnv([\"PATH+MAVEN=\${tool 'mavenInstallationName'}/bin\"]) {"
+        "not-found-pattern, simple-"                | "withMaven(maven: 'mavenInstallationName') {"
+        "this-one-does-not-match, this-one-too"     | "withEnv([\"PATH+MAVEN=\${tool 'mavenInstallationName'}/bin\"]) {"
+        "this-one-does-not-match, this-one-too"     | "withMaven(maven: 'mavenInstallationName') {"
```

Comment:
💭 I guess it does not hurt to test VCS pattern + various Maven install, but wouldn't be enough to have separate coverage (one that test injection + various maven setup, one that test injection + various VCS pattern) ?

Reply:
I just meant that this test tested 2 concerns: vcs patterns and mvn configuration. My remark was just to question wether we need to have a test for those 2 concerns at the same time.